src/mon/MDSMonitor.cc

```diff
@@ -2833,22 +2833,15 @@ void MDSMonitor::maybe_replace_gid(mds_gid_t gid,
     pending_fsmap.promote(sgid, fs, info.rank);
 
     *mds_propose = true;
-  } else if (info.state == MDSMap::STATE_STANDBY_REPLAY) {
-    dout(10) << " failing " << gid << " " << info.addr << " mds." << info.rank << "." << info.inc
-      << " " << ceph_mds_state_name(info.state)
+  } else if (info.state == MDSMap::STATE_STANDBY_REPLAY || 
+             info.state == MDSMap::STATE_STANDBY) {
+    dout(10) << " failing and removing " << gid << " " << info.addr << " mds." << info.rank 
+      << "." << info.inc << " " << ceph_mds_state_name(info.state)
```

Comment:
@gregsfortytwo Yes, I dropped last_beacon.erase(gid) because fail_mds_gid(gid) did this work here. 

I checked git log and this duplicated case was introduced back to 2011. At that time this part made sense because the logic was a little different from now.